src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -235,15 +235,17 @@ public Tensor ProcessImage(string path)
                 ImageClassificationMetrics metrics = new ImageClassificationMetrics();
                 metrics.Bottleneck = new BottleneckMetrics();
                 metrics.Bottleneck.DatasetUsed = dataset;
+                float[] imageArray = null;
                 while (cursor.MoveNext())
                 {
                     labelGetter(ref label);
                     imagePathGetter(ref imagePath);
                     var imagePathStr = imagePath.ToString();
                     var imageTensor = imageProcessor.ProcessImage(imagePathStr);
                     runner.AddInput(imageTensor, 0);
-                    var featurizedImage = runner.Run()[0]; // Reuse memory?
-                    writer.WriteLine(label - 1 + "," + string.Join(",", featurizedImage.ToArray<float>()));
+                    var featurizedImage = runner.Run()[0]; // Reuse memory
+                    featurizedImage.ToArray<float>(ref imageArray);
+                    writer.WriteLine(label - 1 + "," + string.Join(",", imageArray));
```

Comment:
This is potentially wrong because `imageArray.Length` has the potential to be bigger than the actual values stored in `featurizedImage`.

Image 2 rows. One has 1000 elements. The second has 500 elements. When the first one runs, it will create an array of size `1000` and copy all the values into it. On the next pass through the loop, when you get to the row that has `500` elements in it, you will reuse the `1000`-length array from the first loop and only fill up 500 elements in it with the 2nd row of data. But here you are writing out all `1000` elements because you are using every element in the array.

However, if you are guaranteed that every `featurizedImage` through the loop will have the same size, then you should probably add an assert here to catch any bugs.

Reply:
Can I use instead an assert, string join method like this:
public static string Join (string separator, string[] value, int startIndex, int count);

Reply:
You can't use that method because the array is a `string[]` in that method, and you have a `float[]`.

Reply:
I know, but I didn't get any errors and all tests passed when I used that method.